torch/csrc/jit/ir/alias_analysis.cpp

```diff
@@ -1068,7 +1116,9 @@ void AliasDb::createValue(const Value* value) {
   elementMap_[value] = new_elem;
 }
 
-void AliasDb::giveFreshAlias(const Value* value) {
+void AliasDb::giveFreshAlias(
+    const Value* value,
+    bool add_wildcard_to_contained_elems) {
```

Comment:
Seems like sort of a smell that this method grew this flag. It's not clear from the signature alone when it would be appropriate to set it true or false, since it depends heavily on callsite-specific conditions. In general, it is a sign the code is factored oddly when there is a flag to a method which is set for only a single callsite, whereas every other callsite uses a different behavior.

Ideally, the aliasing language encodes these relationships so we don't need to special case the code for it, right?

Reply:
yea, agreed it isn't ideal, see comment above for general discussion of aliasing language respond here for code-specific suggestions/refactorings